src/highlight.js

```diff
@@ -4,692 +4,248 @@ https://highlightjs.org/
 */
 
 // @ts-ignore
-import deepFreeze from 'deep-freeze-es6';
-import Response from './lib/response.js';
-import TokenTreeEmitter from './lib/token_tree.js';
-import * as regex from './lib/regex.js';
-import * as utils from './lib/utils.js';
-import * as MODES from './lib/modes.js';
-import { compileLanguage } from './lib/mode_compiler.js';
-import * as packageJSON from '../package.json';
-import * as logger from "./lib/logger.js";
-import HTMLInjectionError from "./lib/html_injection_error.js";
-
-
-/**
-@typedef {import('highlight.js').Mode} Mode
-@typedef {import('highlight.js').CompiledMode} CompiledMode
-@typedef {import('highlight.js').CompiledScope} CompiledScope
-@typedef {import('highlight.js').Language} Language
-@typedef {import('highlight.js').HLJSApi} HLJSApi
-@typedef {import('highlight.js').HLJSPlugin} HLJSPlugin
-@typedef {import('highlight.js').PluginEvent} PluginEvent
-@typedef {import('highlight.js').HLJSOptions} HLJSOptions
-@typedef {import('highlight.js').LanguageFn} LanguageFn
-@typedef {import('highlight.js').HighlightedHTMLElement} HighlightedHTMLElement
-@typedef {import('highlight.js').BeforeHighlightContext} BeforeHighlightContext
-@typedef {import('highlight.js/private').MatchType} MatchType
-@typedef {import('highlight.js/private').KeywordData} KeywordData
-@typedef {import('highlight.js/private').EnhancedMatch} EnhancedMatch
-@typedef {import('highlight.js/private').AnnotatedError} AnnotatedError
-@typedef {import('highlight.js').AutoHighlightResult} AutoHighlightResult
-@typedef {import('highlight.js').HighlightOptions} HighlightOptions
-@typedef {import('highlight.js').HighlightResult} HighlightResult
-*/
-
-
-const escape = utils.escapeHTML;
-const inherit = utils.inherit;
-const NO_MATCH = Symbol("nomatch");
-const MAX_KEYWORD_HITS = 7;
-
-/**
- * @param {any} hljs - object that is extended (legacy)
- * @returns {HLJSApi}
- */
+import deepFreeze from 'deep-freeze-es6'; // Prevents objects from being modified
```

Comment:
Sorry, I just don't believe in these types of comments.  Comments should explain WHY, not WHAT.

Of course `regex` is going to contain regex utils/stuff... The code explains that already just with the naming, a comment is not necessary - it's just clutter.